controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1358,6 +1360,110 @@ func (r *BareMetalHostReconciler) actionDeprovisioning(prov provisioner.Provisio
+		if hup.Spec.FirmwareSettings == metal3api.HostUpdatePolicyOnReboot {
+			servicingData.LiveFirmwareSettingsAllowed = true
+			info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). hup.Spec.FirmwareSettings set to OnReboot, setting LiveFirmwareSettingsAllowed to  %+v", servicingData.LiveFirmwareSettingsAllowed))
+		}
+		if hup.Spec.FirmwareUpdates == metal3api.HostUpdatePolicyOnReboot {
+			servicingData.LiveFirmwareUpdateAllowed = true
+			info.log.Info(fmt.Sprintf("janders-debug: in checkServicing(). hup.Spec.FirmwareUpdates set to OnReboot, setting LiveFirmwareUpdateAllowed to  %+v", servicingData.LiveFirmwareUpdateAllowed))
+		}
+	}
+
+	if servicingData.LiveFirmwareUpdateAllowed {
+		if !reflect.DeepEqual(info.host.Status.Provisioning.Firmware, info.host.Spec.Firmware) {
+			servicingData.FirmwareConfig = info.host.Spec.Firmware
+			fwDirty = true
+			info.log.Info(fmt.Sprintf("janders-debug: setting fwDirty to  %+v", fwDirty))
+		}
+	}
+
+	if servicingData.LiveFirmwareSettingsAllowed {
+		var hfs *metal3api.HostFirmwareSettings
+		var err error
+		hfsDirty, hfs, err = r.getHostFirmwareSettings(info)
+		info.log.Info(fmt.Sprintf("janders-debug: hfsDirty = %+v, hfs = %+v", hfsDirty, hfs))
+		if err != nil {
+			info.log.Info(fmt.Sprintf("janders-debug: r.getHostFirmwareSettings failed: %q", err))
+			return actionError{fmt.Errorf("could not determine updated settings: %w", err)}, false
+		}
+		if hfsDirty {
+			servicingData.ActualFirmwareSettings = hfs.Status.Settings
+			servicingData.TargetFirmwareSettings = hfs.Spec.Settings
+		}
+	}
+
+	dirty := fwDirty || hfsDirty
+	info.log.Info(fmt.Sprintf("janders-debug: dirty is: %+v", dirty))
+
+	// Even if settings are clean, we need to check the result of the current servicing.
+	if !dirty && info.host.Status.OperationalStatus != metal3api.OperationalStatusServicing && info.host.Status.ErrorType != metal3api.ServicingError {
+		// If nothing is going on, return control to the power management.
+		info.log.Info(fmt.Sprintf("janders-debug: dirty is %+v, info.host.Status.OperationalStatus is %+v, info.host.Status.ErrorType is %+v", dirty, info.host.Status.OperationalStatus, info.host.Status.ErrorType))
+		info.log.Info("janders-debug: nothing happening, return control to the power management (return nil, false)")
+		return nil, false
+	}
+
+	provResult, started, err := prov.Service(servicingData, dirty,
+		info.host.Status.ErrorType == metal3api.ServicingError)
+	if err != nil {
+		info.log.Info(fmt.Sprintf("janders-debug: error getting provResult: %q, false", err))
+		return actionError{fmt.Errorf("error servicing host: %w", err)}, false
+	}
+	if provResult.ErrorMessage != "" {
+		result = recordActionFailure(info, metal3api.ServicingError, provResult.ErrorMessage)
+		info.log.Info(fmt.Sprintf("janders-debug: error getting provResult: %+v, false", result))
+		return result, true
+	}
+	if started && clearErrorWithStatus(info.host, metal3api.OperationalStatusServicing) {
```

Comment:
Before this if started I would add the logic to handle the update for HFC information, for reference you can look at https://github.com/metal3-io/baremetal-operator/blob/main/controllers/metal3.io/baremetalhost_controller.go#L1178-L1190